locopy/snowflake.py

```diff
@@ -424,20 +424,36 @@ def insert_dataframe_to_table(
             If metadata==None, it will be generated based on data
         """
         import pandas as pd
+        import polars as pl
+        import polars.selectors as cs
 
         if columns:
-            dataframe = dataframe[columns]
+            try:
```

Comment:
Looking at the method, would it be possible to use a single dispatch helper here for the first part of this function? E.g. a dispatched function for line 431 to 456 - creating the `to_insert`. Once we have the list of tuples, the rest is the same.

Reply:
Yes it is possible, my only concern was readability for future developers, but happy to refactor it to dispatched function if there is any significant upside

Reply:
I think it's more readable to use the isinstance checks below to select columns from dataframe. I don't think anything between requires this to be done first. That or doing the dispatch. Lmk if I'm wrong here. Like moving the `dataframe=dataframe[columns]` into the pandas branch and the `.select(columns)` into the polars branch.

Reply:
Actually polars dataframe can read `dataframe[columns]`, the `.select(columns)` was intended for LazyFrame. Since we are not supporting LazyFrame now, it can be removed completely

Reply:
Sounds good to me